pkg/plugins/policies/core/rules/resourcerules.go

```diff
@@ -254,13 +263,36 @@ func resolveTargetRef(item PolicyItemWithMeta, reader ResourceReader) []*resolve
 	rtype := core_model.ResourceType(item.GetTargetRef().Kind)
 	list := reader.ListOrEmpty(rtype).GetItems()
 
-	if len(item.GetTargetRef().Labels) > 0 {
+	var implicitPort uint32
+	implicitLabels := map[string]string{}
+	if item.GetTargetRef().Kind == common_api.MeshService {
+		if name, namespace, port, err := parseService(item.GetTargetRef().Name); err == nil {
```

Comment:
don't we need to do this only if port (the section name?) is missing?

Reply:
yeah